plugins/ai_assistant.py

```diff
@@ -0,0 +1,107 @@
+import random
+import string
+import json
+import textwrap
+from base64 import b64decode
+
+from agents import (
+    Agent,
+    Runner,
+    WebSearchTool,
+    ImageGenerationTool,
+    FunctionTool,
+    function_tool,
+    set_default_openai_key,
+    TResponseInputItem
+)
+
+from google.cloud import storage
+
+from cloudbot import hook
+from cloudbot.bot import bot
+from cloudbot.util import web
+
+set_default_openai_key(bot.config.get_api_key("openai"))
+
+CONTEXT: list[TResponseInputItem] = []
+
+@function_tool()
+@hook.command("dump_context", autohelp=False)
+async def dump_context():
+    """
+    dump_context deletes the current context and history of the conversation.
+    """
+    global CONTEXT
+    CONTEXT = []
+
+    return "The context has been cleared."
+
+
+web_agent = Agent(
+    name="Web Assistant",
+    instructions="Be a helpful assistant. You can search the web for information.",
+    handoff_description="Use this agent for searching the web for information.",
+    tools=[
+        WebSearchTool(),
+    ]
+)
+
+default_agent = Agent(
+    name="Gloria",
+    instructions="Be a helpful assistant.",
+    tools=[
+        web_agent.as_tool(
+             tool_name="web_search",
+             tool_description="Search the web for information and provide relevant results."
+        ),
+        ImageGenerationTool(tool_config={
+            "type": "image_generation",
+            "model": "gpt-image-1",
+            "size": "1024x1024"
+        }),
+        dump_context
+    ]
+)
+
```

Comment:
```suggestion
default_agent = Agent(
    name="Gloria",
    instructions="Be a helpful assistant.",
    tools=[
        Agent(
            name="Web Assistant",
            instructions="Be a helpful assistant. You can search the web for information.",
            handoff_description="Use this agent for searching the web for information.",
            tools=[
                WebSearchTool(),
             ]).as_tool(
               tool_name="web_search",
               tool_description="Search the web for information and provide relevant results."
         ),
        ImageGenerationTool(tool_config={
            "type": "image_generation",
            "model": "gpt-image-1",
            "size": "1024x1024"
        }),
        dump_context
    ]
)

```

Not sure if it works, but it would be clearer to read this way ?